### DISCUSSION TITLE
Can this module support file tree like monaco-vscode-api?

### DISCUSSION THREAD

**Discussion by invm** [TypeFox/monaco-languageclient#716] [Q&A] [answered]
I came here after seeing this [issue](https://github.com/TypeFox/monaco-languageclient/issues/711),

Do you think this is achievable?

**Comment by CGNonofr** [comment, marked as answer]
This package relies on `monaco-vscode-api` so I don't see any reason why not

I just have the feeling that if you need something THAT advanced that you want the whole VSCode workbench, you may probably want to use `monaco-vscode-api` directly

**Reply by invm** [reply]
The demo on monaco-vscode-api does not have anything regarding language client, that why I wanted to use this module, any reference how add the language client similar to examples in this repo?

**Reply by CGNonofr** [reply]
You can still use the monaco-languageclient `MonacoLanguageClient` (but [it doesn't do much](https://github.com/TypeFox/monaco-languageclient/blob/84cd57ebad99bfe019ed010010e0639ad93f77bb/packages/client/src/index.ts#L19C14-L30)) or the vscode-languageclient directly, like you would do in a VSCode extension

**Reply by invm** [reply]
Yeah I assumed that from your previous answer, but how to actually do it? I see mentions of the language client in the readme but how to actually plug it in? It really seems easier from this module simply because it is documented.

What would be the benefit to using monaco-vscode-api instead of this module? Will my app be lighter? My end goal is to have minimal complexity rather than bundle size footprint.

**Reply by CGNonofr** [reply]
> Yeah I assumed that from your previous answer, but how to actually do it? I see mentions of the language client in the readme but how to actually plug it in? It really seems easier from this module simply because it is documented.

The goal of monaco-vscode-api is to be able to run VSCode extensions, there are some specificities (like the possibility to access the extension api directly from the main thread) but you can mainly rely on the [official VSCode documentation](https://code.visualstudio.com/api/language-extensions/language-server-extension-guide)

> What would be the benefit to using monaco-vscode-api instead of this module? Will my app be lighter? My end goal is to have minimal complexity rather than bundle size footprint.

monaco-languageclient relies on monaco-vscode-api, it's a kind a wrapper designed to simplify its usage, but it also limit the possibilities for advanced usecases

If you tell me more about your exact usecase, I can maybe suggest what I think is the better option for you

**Reply by invm** [reply]
I'm building a desktop app with tauri where I plan on working with 2 language servers and a minimal editor, I only plan on file system provider and git, everything else will be stripped, I do not need extensions.

I assume that this module will be easier to implement and maintain in that case.

**Reply by CGNonofr** [reply]
I see, I hope you are aware that the bundled services are the one for the web version of VSCode, so you may encounter some issues trying to use them in a native environment

**Reply by invm** [reply]
I assume I'll need to monitor fs events for files changes and git changes manually and manually start language servers, which is easily doable with tauri, I think for my minimal requirements that should suffice.

As always, thanks for being super responsive.
